File: Competitive Programming 3/Section 1.4/Game (Card)/11225 - Tarot scores/Java/Main.java
import java.util.Scanner;
import java.util.HashMap;
import java.util.HashSet;
import java.text.DecimalFormat;

public class Main {
    static Scanner s = new Scanner(System.in);
    static HashMap<String,Integer> mapToPoints = new HashMap<String,Integer>(){{
        put("king",9);
        put("queen",7);
        put("knight",5);
        put("jack",3);
    }};
    static HashSet<String> mapToPointsOudlers = new HashSet<String>(){{
        add("fool");
        add("one of trumps");
        add("twenty-one of trumps");
    }};
    static int[] pointsToWin = {112,102,82,72};
    public static void main(String[]args){
        int t = Integer.parseInt(s.nextLine());
        for(int cs=1;cs<=t;cs++){
            if(cs>1)
                System.out.printf("\n");
            System.out.printf("Hand #%d\n",cs);
            int points=0,oudlers=0;
            for(int n = Integer.parseInt(s.nextLine());n > 0;n--){
                String card=s.nextLine();
                String begCard = card.split(" ")[0];
                //check if it is oudler
                if(mapToPointsOudlers.contains(card)){
                    points+=9;
                    oudlers++;
                }else{
                    //sum points
                    Integer p = mapToPoints.get(begCard);
                    if(p==null){
                        points++;
                    }else{
                        points+=p;
                    }
                }
            }
            int dif = Math.abs(pointsToWin[oudlers]-points);
            DecimalFormat format = new DecimalFormat("0.##");
            System.out.printf("Game %s by %s point(s).\n",points<pointsToWin[oudlers]?"lost":"won",format.format((float)dif/2));
        }
    }
}

File: Competitive Programming 3/Section 1.4/Game (Card)/11225 - Tarot scores/Python/main.py
mapToPoints={
    "king":9,
    "queen":7,
    "knight":5,
    "jack":3
}
mapToPointsOudlers={
    "fool",
    "one of trumps",
    "twenty-one of trumps"
}
pointsToWin=[112,102,82,72]
def main():
    t = int(input())+1
    for cs in range(1,t):
        if cs > 1:
            print()
        print("Hand #"+str(cs))
        points,oudlers=0,0
        n = int(input())
        for _ in range(0,n):
            card=input()
            begCard = card.split()[0]
            #check if it is oudler
            if card in mapToPointsOudlers:
                points+=9
                oudlers+=1
            else:
                #sum points
                if begCard in mapToPoints:
                    points+=mapToPoints[begCard]
                else:
                    points+=1
        dif = abs(pointsToWin[oudlers]-points)
        print("Game "+("lost" if points<pointsToWin[oudlers] else "won")+" by "+str(int(dif/2) if dif%2==0 else str(dif/2))+" point(s).")
main()